packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,852 @@
+    from: string,
+    to: string,
+    resourceAddress: string,
+    amount: number,
+    amountToLockForFees: number,
+  ): TransactionManifest {
+    return new ManifestBuilder()
+      .callMethod(from, 'lock_fee', [decimal(amountToLockForFees)])
+      .callMethod(from, 'withdraw', [address(resourceAddress), decimal(amount)])
+      .takeFromWorktop(resourceAddress, decimal(amount).value, (builder, bucketId) => {
+        return builder.callMethod(to, 'try_deposit_or_abort', [bucket(bucketId), enumeration(0)])
+      })
+      .build()
+  }
+
+  private async constructIntent(
+    manifest: TransactionManifest,
+    message: Message,
+    nonce: number,
+    notaryPublicKey: PublicKey,
+  ): Promise<Intent> {
+    const epoch = await this.currentEpoch()
+    return {
+      header: {
+        networkId: this.networkId,
+        startEpochInclusive: epoch,
+        endEpochExclusive: epoch + 10,
+        nonce,
+        notaryPublicKey,
+        notaryIsSignatory: true,
+        tipPercentage: 0,
+      },
+      manifest,
+      message,
+    }
+  }
+
+  private async previewIntent(intent: Intent): Promise<TransactionPreviewResponse> {
+    // Translate the RET models to the gateway models for preview.
+    const request: TransactionPreviewOperationRequest = {
+      transactionPreviewRequest: {
+        manifest: await RadixEngineToolkit.Instructions.convert(
+          intent.manifest.instructions,
+          this.networkId,
+          'String',
+        ).then((instructions) => instructions.value as string),
+        blobs_hex: [],
+        start_epoch_inclusive: intent.header.startEpochInclusive,
+        end_epoch_exclusive: intent.header.endEpochExclusive,
+        notary_public_key: RadixSpecificClient.retPublicKeyToGatewayPublicKey(intent.header.notaryPublicKey),
+        notary_is_signatory: intent.header.notaryIsSignatory,
+        tip_percentage: intent.header.tipPercentage,
+        nonce: intent.header.nonce,
+        signer_public_keys: [RadixSpecificClient.retPublicKeyToGatewayPublicKey(intent.header.notaryPublicKey)],
+        // TODO: Add message
+        flags: {
+          assume_all_signature_proofs: false,
+          skip_epoch_check: false,
+          use_free_credit: false,
+        },
+      },
+    }
+
+    return this.innerGatewayClient.transaction.innerClient.transactionPreview(request)
+  }
+
+  private static retPublicKeyToGatewayPublicKey(publicKey: PublicKey): GatewayPublicKey {
+    switch (publicKey.curve) {
+      case 'Secp256k1':
+        return {
+          key_type: 'EcdsaSecp256k1',
+          key_hex: publicKey.hex(),
+        }
+      case 'Ed25519':
+        return {
+          key_type: 'EddsaEd25519',
+          key_hex: publicKey.hex(),
+        }
+    }
+  }
+  // #endregion Private Methods
+}
+
+/**
+ * Custom Radix client
+ */
+
+export default class Client extends BaseXChainClient {
+  private radixSpecificClient: RadixSpecificClient
+  private curve: Curve
+  constructor(
```

Comment:
To be consistent with the rest of XChainJS clients, use the following type for the constructor `XChainClientParams & { curve?: Curve }`. To simplify the creation of the client, I would add 'Ed25519' as default value to the curve parameter.

Reply:
done